lib/Sema/TypeCheckDecl.cpp

```diff
@@ -7022,6 +7022,13 @@ static Optional<ObjCReason> shouldMarkClassAsObjC(TypeChecker &TC,
 
   if (auto attr = CD->getAttrs().getAttribute<ObjCAttr>()) {
     if (kind == ObjCClassKind::ObjCMembers) {
```

Comment:
After getting confused by this, I feel a deep desire to rename `ObjCClassKind::ObjCMembers` to `ObjCClassKind::ObjCWithGenericAncestry`, especially given that `ObjCMembers` is no longer even true for Swift 4. Anyway, not critical to your change.